src/System.IO.FileSystem/src/System/IO/PathHelpers.Unix.cs

```diff
@@ -27,5 +27,12 @@ internal static void CheckSearchPattern(string searchPattern)
                 }
             }
         }
+
+        internal static string TrimEndingDirectorySeparator(string path)
+        {
+            return path.Length > 1 && EndsInDirectorySeparator(path) ? // exclude root "/"
```

Comment:
nano optimization, but `EndsInDirectorySeparator` performs a null/empty check which is unnecessary since your methods assume that. You could just do `IsDirectorySeparator(path[path.Length - 1])`